### ISSUE TITLE
Proposal: Group pull requests

### ISSUE THREAD

**Issue by terryhendrix1990** [scala-steward-org/scala-steward#2705]
**Context** 

A multitude of dependency update PR's are created when scala steward runs. After merging one of the PR's a new CI build  (for the other PR's) is needed  in case:
1. Merge conflicts occur.
2. PR's need to be up to date with master, before merging to master.

When SBT only builds delta's (changes since last successful build) then even more PR's are created. I see this happens because `currentVersion` or `newerVersions` is different when the `groupId` is shared between dependencies. This causes `groupByGroupId` and `groupByArtifactIdName` to return multiple Groups for the same `groupId`, thus creating more PR's.

All of the above creates an additional workload, which I'd like to solve.

**Proposal**

I propose to add another kind of grouping to scala-steward, namely PR grouping. The general idea to add new settings to scala-steward which determines how PR's should be grouped, this will lead to less PR's, thus reducing the additional workload.

The configuration could look like:

```
groupPullRequests = [
    {["io.abc", "io.def", "io.ghi" ]},    // Group these unrelated groups in a single PR
    {["io.mycompany.*" ]},                // Group all dependencies of the company you work for in a single PR.
    {["*"]},                              // Group the remaining dependencies in a single PR
]
```

If the wildcard group `{["*"]}` is absent, then the remaining `UpdateData`s get their own PR, which is the current default in scala steward.

Then for PR grouping: I think it could be achieved by making `UpdateData` a trait, which is implemented in `UpdateData.Single` and a `UpdateData.Group`. This is similar to how grouping for `Update` works. 

I'd like to do (or assist in) the implementation. But first things first:
I'd love to get some feedback from the maintainers/scala-steward experts. Do you think this could end up in the main-branch?

**Comment by alejandrohdezma** [comment]
Hey @terryhendrix1990. I've just started working on #2714 and saw your issue while looking if this approach have already been tried in the past. Since both ideas are similar, would you like to work together on it?

**Comment by terryhendrix1990** [comment]
Hi @alejandrohdezma 
That sounds great, how do we proceed?

**Comment by alejandrohdezma** [comment]
Feel free to ping me at either my email or Twitter if you want (both are in my profile) to ease the technical discussion. Nevertheless if you prefer to keep it in here I'm fine with that too 😸

**Comment by terryhendrix1990** [comment]
I've send you a message on Twitter

**Comment by exoego** [comment]
I think this is already implemented in #2714.
Feel free to reopen new issue if you want to add more.
